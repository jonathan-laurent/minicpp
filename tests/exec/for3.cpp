#include <iostream>

int main() {
  int i;
  int j;
  for (i = 0; i < 5; i++)
    for (j = 0; j < 5; j++)
      std::cout << "i*j = " << i*j << "\n";
  std::cout << "i = " << i << "\n";
  std::cout << "j = " << j << "\n";
}
