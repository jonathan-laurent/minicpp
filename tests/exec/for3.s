	.text
	jal  _p0__main
	li   $v0, 10
	syscall
_p0__main:
	addi $sp, $sp, -4
	li   $t1, 0
L57:
	li   $t2, 5
	slt  $t2, $t1, $t2
	bnez $t2, L53
	la   $a0, _s__3
	li   $v0, 4
	syscall
	move $a0, $t1
	li   $v0, 1
	syscall
	la   $a0, _s__2
	li   $v0, 4
	syscall
	la   $a0, _s__5
	li   $v0, 4
	syscall
	move $a0, $t0
	li   $v0, 1
	syscall
	la   $a0, _s__4
	li   $v0, 4
	syscall
	addi $sp, $sp, 4
	jr   $ra
L53:
	li   $t0, 0
L51:
	li   $t2, 5
	slt  $t2, $t0, $t2
	bnez $t2, L47
	addi $t1, $t1, 1
	b    L57
L47:
	la   $a0, _s__1
	li   $v0, 4
	syscall
	mul  $a0, $t1, $t0
	li   $v0, 1
	syscall
	la   $a0, _s__0
	li   $v0, 4
	syscall
	addi $t0, $t0, 1
	b    L51
	.data
_s__0:
	.asciiz "\n"
_s__1:
	.asciiz "i*j = "
_s__2:
	.asciiz "\n"
_s__3:
	.asciiz "i = "
_s__4:
	.asciiz "\n"
_s__5:
	.asciiz "j = "

