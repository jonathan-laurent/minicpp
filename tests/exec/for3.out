i*j = 0
i*j = 0
i*j = 0
i*j = 0
i*j = 0
i*j = 0
i*j = 1
i*j = 2
i*j = 3
i*j = 4
i*j = 0
i*j = 2
i*j = 4
i*j = 6
i*j = 8
i*j = 0
i*j = 3
i*j = 6
i*j = 9
i*j = 12
i*j = 0
i*j = 4
i*j = 8
i*j = 12
i*j = 16
i = 5
j = 5
